### ISSUE TITLE
Problem compiling aubio.

### ISSUE THREAD

**Issue by garry-iglesias** [mxe/mxe#1374]
Hi,
I've been using mxe for several years, but I got a recurrent problem for a few month, the full make stop when compiling aubio.

```
Failed to build package aubio for target i686-w64-mingw32.static!
------------------------------------------------------------
/tmp/ccBCdp1N.o:aubio-test.c:(.text+0x82): undefined reference to `fvec_print'
/tmp/ccBCdp1N.o:aubio-test.c:(.text+0x8e): undefined reference to `del_fvec'
collect2: error: ld returned 1 exit status
```

I'm not sure but I got the impression that "one time" (I do daily fetch from repo) it had worked within the period, but it was exceptional.

Sorry I don't have time to track this compilation problem more in depth, and I work around the problem compiling specifically what I need, but if we could "blind make" that would be great.

**Comment by spthaolt** [comment]
Hello,

I'm having the same problem too.
My system specs. is:

CentOS 7 x86_64
GCC 4.8.5

**Comment by spthaolt** [comment]
Maybe you could try this patch, it works for me at least

``` diff
diff --git a/src/aubio.mk b/src/aubio.mk
index df05acb..0a32734 100644
--- a/src/aubio.mk
+++ b/src/aubio.mk
@@ -28,6 +28,7 @@ define $(PKG)_BUILD
             --with-target-platform='win$(BITS)'   \
             --prefix='$(PREFIX)/$(TARGET)'        \
             --enable-fftw3f                       \
+            --libdir='$(PREFIX)/$(TARGET)/lib'    \
             $(if $(BUILD_STATIC),                 \
                 --enable-static --disable-shared --disable-jack, \
                 --disable-static --enable-shared)
```

**Comment by garry-iglesias** [comment]
Thank you.

Honestly it was not blocking for me at the time, I just wanted to give the
info back, because other people could need it, and because I felt annoying
to not be able to compile all "without a scratch" :).

**Comment by starius** [comment]
@spthaolt 
Can you provide build logs before and after applying the patch, please?
You can find them in `log/*/aubio_i686-w64-mingw32.static`

**Comment by tonytheodore** [comment]
An [issue template](https://github.com/mxe/mxe/issues/1235) would help this greatly. Without more details and logs, we can't see what the problem is. MXE has some known issues with certain platforms, but CentOS isn't one of them.

**Comment by spthaolt** [comment]
Hi,

I have attached the failed log file here: 
[aubio_i686-w64-mingw32.static.txt](https://github.com/mxe/mxe/files/385160/aubio_i686-w64-mingw32.static.txt)

This was the error message on the screen:
```
[build]     aubio                    i686-w64-mingw32.static                                             

Failed to build package aubio for target i686-w64-mingw32.static!
------------------------------------------------------------
/tmp/ccXqGgwb.o:aubio-test.c:(.text+0x54): undefined reference to `fvec_zeros'
/tmp/ccXqGgwb.o:aubio-test.c:(.text+0x82): undefined reference to `fvec_print'
/tmp/ccXqGgwb.o:aubio-test.c:(.text+0x8e): undefined reference to `del_fvec'
collect2: error: ld returned 1 exit status
make[1]: *** [build-only-aubio_i686-w64-mingw32.static] Error 1
make[1]: Leaving directory `/home/thaolt/Workspaces/mxe'
real	0m15.502s
user	0m16.472s
sys	0m1.870s
------------------------------------------------------------
[log]      /home/thaolt/Workspaces/mxe/log/aubio_i686-w64-mingw32.static
```

I checked the log file and saw the detail error of pkgconfig
```
[...]
'i686-w64-mingw32.static-gcc' -W -Wall -Werror -ansi -pedantic './src/aubio-test.c' -o '/home/thaolt/Workspaces/mxe/usr/i686-w64-mingw32.static/bin/test-aubio.exe' `'i686-w64-mingw32.static-pkg-config' aubio --cflags --libs`
Package aubio was not found in the pkg-config search path.
Perhaps you should add the directory containing `aubio.pc'
to the PKG_CONFIG_PATH environment variable
Package 'aubio', required by 'world', not found
/tmp/ccXqGgwb.o:aubio-test.c:(.text+0x16): undefined reference to `new_fvec'
[...]
```

The log reported that aubio lib was compiled and installed successfully. The last step was to compile aubio-test from mxe src/ and it seems pkgconfig cannot find aubio.pc

aubio.pc was installed into mxe/usr/i686-w64-mingw32.static/lib64/pkgconfig/aubio.pc

```
[...]
+ install /home/thaolt/Workspaces/mxe/usr/i686-w64-mingw32.static/lib64/pkgconfig/aubio.pc (from build/aubio.pc)
[...]
```

First, I try to export env variable of PKG_CONFIG_PATH -> not work
Next, edited aubio.mk to add --define-variable=pc_path=[...]/lib64/pkgconfig -> not work too
Then I saw other packages install their *.pc files in lib/ instead of lib64/, so I edited aubio.mk to change the configuration of aubio install libdir. 

I don't know whether the change would have effect on other packages or not. The building is still running it's 10 hours since I started :)

**Comment by tonytheodore** [comment]
> Then I saw other packages install their *.pc files in lib/ instead of lib64/, so I edited aubio.mk to change the configuration of aubio install libdir.

Nice investigation - thanks! Could you raise a PR for this?

**Comment by spthaolt** [comment]
Hi again,

I have just created a pull request. I don't know if I did it correctly. Sorry, I'm not a github experienced user.

**Comment by tonytheodore** [comment]
fixed in #1459
